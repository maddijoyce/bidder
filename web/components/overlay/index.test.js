import React from 'react';
import Renderer from 'react-test-renderer';

import Overlay from './index';

test('Overlay uses defaults', () => {
  const component = Renderer.create(
    <Overlay />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Overlay uses error code and message', () => {
  const component = Renderer.create(
    <Overlay
      code="404"
      title="This is an error"
      detail="This is more info"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
