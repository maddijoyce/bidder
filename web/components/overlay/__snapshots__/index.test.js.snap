// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay uses defaults 1`] = `
<div
  className="overlay"
>
  <div
    className="overlay-content"
  >
    <h6
      className="overlay-code"
    >
      Error
    </h6>
    <hr
      className="overlay-rule"
    />
    <h1
      className="overlay-title"
    >
      Something's gone wrong
    </h1>
    <hr
      className="overlay-rule"
    />
    <p
      className="overlay-detail"
    >
      <span>
        If you'd like to 
        <a
          href="mailto:error@bidder.maddijoyce.com"
        >
          let us know
        </a>
        , we'll see what we can do to fix it.
      </span>
    </p>
  </div>
</div>
`;

exports[`Overlay uses error code and message 1`] = `
<div
  className="overlay"
>
  <div
    className="overlay-content"
  >
    <h6
      className="overlay-code"
    >
      404
    </h6>
    <hr
      className="overlay-rule"
    />
    <h1
      className="overlay-title"
    >
      This is an error
    </h1>
    <hr
      className="overlay-rule"
    />
    <p
      className="overlay-detail"
    >
      This is more info
    </p>
  </div>
</div>
`;
